exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/WindowFunction.kt

```diff
@@ -0,0 +1,456 @@
+        orderBy(column to order)
+
+    /**
+     * Defines sorting order by column and order pairs [order] in which the rows of a partition
+     * are processed by the window function.
+     */
+    fun orderBy(vararg order: Pair<Expression<*>, SortOrder>): WindowFunctionDefinition<T> = apply {
+        (orderByExpressions as MutableList).addAll(order)
+    }
+
+    /**
+     * Defines the set of rows constituting the window frame, which is a subset of the current partition.
+     * Window frame uses [WindowFrameUnit.ROWS] mode and specified [start] and [end] bounds.
+     */
+    fun rows(
+        start: WindowFrameBound,
+        end: WindowFrameBound
+    ): WindowFunctionDefinition<T> = apply {
+        frameClause = WindowFrameClause(WindowFrameUnit.ROWS, start, end)
+    }
+
+    /**
+     * Defines the set of rows constituting the window frame, which is a subset of the current partition.
+     * Window frame uses [WindowFrameUnit.ROWS] mode and specified [start] bound.
+     */
+    fun rows(
+        start: CurrentOrPreceding
+    ): WindowFunctionDefinition<T> = apply {
+        frameClause = WindowFrameClause(WindowFrameUnit.ROWS, start, null)
+    }
+
+    /**
+     * Defines the set of rows constituting the window frame, which is a subset of the current partition.
+     * Window frame uses [WindowFrameUnit.RANGE] mode and specified [start] and [end] bounds.
+     */
+    fun range(
+        start: WindowFrameBound,
+        end: WindowFrameBound
+    ): WindowFunctionDefinition<T> = apply {
+        frameClause = WindowFrameClause(WindowFrameUnit.RANGE, start, end)
+    }
+
+    /**
+     * Defines the set of rows constituting the window frame, which is a subset of the current partition.
+     * Window frame uses [WindowFrameUnit.RANGE] mode and specified [start] bound.
+     */
+    fun range(
+        start: CurrentOrPreceding
+    ): WindowFunctionDefinition<T> = apply {
+        frameClause = WindowFrameClause(WindowFrameUnit.RANGE, start, null)
+    }
+
+    /**
+     * Defines the set of rows constituting the window frame, which is a subset of the current partition.
+     * Window frame uses [WindowFrameUnit.GROUPS] mode and specified [start] and [end] bounds.
+     */
+    fun groups(
+        start: WindowFrameBound,
+        end: WindowFrameBound
+    ): WindowFunctionDefinition<T> = apply {
+        frameClause = WindowFrameClause(WindowFrameUnit.GROUPS, start, end)
+    }
+
+    /**
+     * Defines the set of rows constituting the window frame, which is a subset of the current partition.
+     * Window frame uses [WindowFrameUnit.GROUPS] mode and specified [start] bound.
+     */
+    fun groups(
+        start: CurrentOrPreceding,
+    ): WindowFunctionDefinition<T> = apply {
+        frameClause = WindowFrameClause(WindowFrameUnit.GROUPS, start, null)
+    }
+
+    override fun toQueryBuilder(queryBuilder: QueryBuilder) = queryBuilder {
+        function.toQueryBuilder(this)
+        +" OVER("
```

Comment:
Thank you for your submission. Could you please run a linter on this code, though?

Reply:
I ran detekt linter and fixed warnings in new code. Although it sees no problem on that particular line.